### ISSUE TITLE
Separate KomaMRI into simpler subpackages

### ISSUE THREAD

**Issue by cncastillo** [cncastillo/KomaMRI.jl#92]
This needs to be done to remove a lot of the dependencies of the UI like Blink, PlotlyJS, etc.

The idea of doing this is to reduce the number of dependencies to the minimum if someone just wants to simulate data.

To replace MRIReco, the KomaMRIBase package should use instead:
  **MRIBase** 
  -> RawAcquisitionData
  -> AcquisitionData
  **MRIFiles**
  -> ISMRMRD

Then, the KomaMRI package can have other dependencies.

Not sure if the plots for our datatypes should be included in KomaMRIBase. Maybe also have KomaMRIPlots?

In summary:
├── KomaMRI.jl (depends on Base and UI)
│   ├── KomaMRIBase.jl (least dependencies)
│   └── KomaMRIPlots.jl (separated to include plots only if needed)

We will need to think about the best way to do this.

------
Subpackages:

- [x] #265
- [x] KomaMRICore
- [x] #232
- [x] KomaMRIPlots

**Comment by cncastillo** [comment]
We need to think what goes where: KomaMRI (UI, All the rest); KomaMRIBase (simulation); KomaMRIPlots (Plots)
ArgCheck = "2" <---Used, but not neccessary
AssetRegistry = "0.1" <---UI
Blink = "0.12" <---UI
CUDA = "3" <---Base
Distributions = "0.25" <---Base
FFTW = "1" <---(?)
FileIO = "1" <---IO
HDF5 = "0.16" <---IO
Hwloc = "2" <---N threads
Images = "0.24, 0.25" <---(?)
Interact = "0.10" <---UI
Interpolations = "0.13, 0.14" <--Base
Ipopt = "0.7, 1" <---Example DiffPrepMCMRF
JLD2 = "0.4" <---IO
JuMP = "0.21, 1" <---Example DiffPrepMCMRF
LinearOperators = "2" <---(?)
MAT = "0.10" <---IO
MRIReco = "0.4.2" <---Recon
MathOptInterface = "0.9, 0.10, 1" <---Example DiffPrepMCMRF
Mux = "0.7" <---UI
PackageCompiler = "1, 2" <---Not used
Parameters = "0.12" <---Base
PlotlyJS = "0.18" <---Plots/UI
ProgressMeter = "1" <---Base
Reexport = "1" <---MRIReco/Blink
Scanf = "0.5" <---Example DiffPrepMCMRF
Suppressor = "0.2" <--Not used

**Comment by beorostica** [comment]
(needed)  CUDA = "3" <---Base
(needed)  Hwloc = "2" <---Base
(needed)  Interpolations = "0.13, 0.14" <-- Base, (por sacar)
(added)   MRIFiles <---Base
(needed)  Parameters = "0.12" <---Base
(needed)  ProgressMeter = "1" <---Base
(needed)  Reexport = "1" <---Base
(added)   ThreadsX <--- Base
(needed)  PlotlyJS = "0.18" <---Base (Plots), UI

(needed)  FileIO = "1" <---Base, IO
(needed)  HDF5 = "0.16" <---Base, IO
(needed)  JLD2 = "0.4" <---Base, IO
(needed)  MAT = "0.10" <---Base, IO
(needed)  Scanf = "0.5" <---Base, IO

(needed)  AssetRegistry = "0.1" <---UI
(needed)  Blink = "0.12" <---UI
(needed)  Interact = "0.10" <---UI
(needed)  MRIReco = "0.4.2" <---Reco, UI

**Comment by cncastillo** [comment]
For the moment, it was decided to separate the package in:

├── KomaMRI.jl: includes the KomaUI and Reconstruction (dependencies: Blink, Mux, MRIReco, KomaMRICore, KomaMRIPlots, etc).
│ ├── **KomaMRICore**.jl: least amount of dependencies, just simulation, and IO: Pulseq, ISMRMRD, etc.
│ └── **KomaMRIPlots**.jl: just the Plots, mainly to remove PlotlyJS as a dependency as it could be a heavy package.

@beorostica Please use these names. All the KomaUI.jl stuff is part of KomaMRI.jl, not KomaMRIPlots. KomaMRIPlots should include the functions defined in DisplayFunctions.jl. For the time being, I would focus on KomaMRICore.

**Comment by cncastillo** [comment]
I re-open this issue as separating the IO part is necessary @beorostica.
